src/ipa/rpi/controller/rpi/sync.cpp

```diff
@@ -0,0 +1,349 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2024, Raspberry Pi Ltd
+ *
+ * sync.cpp - sync algorithm
+ */
+#include "sync.h"
+
+#include <chrono>
+#include <ctype.h>
+#include <fcntl.h>
+#include <list>
+#include <map>
+#include <strings.h>
+#include <unistd.h>
+#include <vector>
+
+#include <libcamera/base/log.h>
+
+#include <arpa/inet.h>
+
+#include "sync_status.h"
+
+using namespace std;
+using namespace std::chrono_literals;
+using namespace RPiController;
+using namespace libcamera;
+
+LOG_DEFINE_CATEGORY(RPiSync)
+
+#define NAME "rpi.sync"
+
+const char *kDefaultGroup = "239.255.255.250";
+constexpr unsigned int kDefaultPort = 10000;
+constexpr unsigned int kDefaultSyncPeriod = 30;
+constexpr unsigned int kDefaultReadyFrame = 1000;
+constexpr unsigned int kDefaultMinAdjustment = 50;
+constexpr unsigned int kDefaultFitNumPts = 100;
+constexpr unsigned int kDefaultFitMaxJitter = 500;
+constexpr unsigned int kDefaultFitMinPts = 10;
```

Comment:
Although libcamera now uses this ``k`` notation, none of our other algorithms do.  Not really fussed either way.

Reply:
Based on the recent AWB patches, I sense leaving the `k`s in may be preferable...